command/ca/provisioner/webhook/update.go

```diff
@@ -0,0 +1,171 @@
+package webhook
+
+import (
+	"errors"
+
+	"github.com/smallstep/certificates/ca"
+	"github.com/smallstep/cli/flags"
+	"github.com/smallstep/cli/utils"
+	"github.com/urfave/cli"
+	"go.step.sm/cli-utils/errs"
+	"go.step.sm/linkedca"
+)
+
+func updateCommand() cli.Command {
+	return cli.Command{
+		Name:   "update",
+		Action: cli.ActionFunc(updateAction),
+		Usage:  "update a webhook attached to a provisioner",
+		UsageText: `**step ca provisioner webhook update** <name> **--provisioner**=<name>
+[**--url**=<url>] [**--kind**=<kind>] [**--bearer-token-file**=<filename>]
+[**--basic-auth-username**=<username>] [**--basic-auth-password-file**=<filename>]
+[**--disable-tls-client-auth**] [**--cert-type**=<cert-type>]
+[**--admin-cert**=<file>] [**--admin-key**=<file>] [**--admin-provisioner**=<name>]
+[**--admin-subject**=<subject>] [**--password-file**=<file>] [**--ca-url**=<uri>]
+[**--root**=<file>] [**--context**=<name>] [**--ca-config**=<file>]`,
+		Flags: []cli.Flag{
+			// General webhook flags
+			provisionerFlag,
+			urlFlag,
+			kindFlag,
+			bearerTokenFileFlag,
+			basicAuthUsernameFlag,
+			basicAuthPasswordFileFlag,
+			disableTLSClientAuthFlag,
+			certTypeFlag,
+
+			flags.AdminCert,
+			flags.AdminKey,
+			flags.AdminProvisioner,
+			flags.AdminSubject,
+			flags.PasswordFile,
+			flags.CaURL,
+			flags.Root,
+			flags.Context,
+			flags.CaConfig,
+		},
+		Description: `**step ca provisioner webhook update** updates a webhook attached to a provisioner.
+
+## POSITIONAL ARGUMENTS
+
+<name>
+: The name of the webhook.
+
+## EXAMPLES
+
+Change a webhook's url:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --url https://example.com
+'''
+
+Configure a webhook to send a bearer token to the server:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --bearer-token-file token.txt
+'''
+
+Change the password sent to the webhook with basic authentication:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --basic-auth-password-file my_pass.txt
+'''
+
+Configure the webhook to be called only when signing x509 certificates, not SSH certificates:
+'''
+step ca provisioner webhook update my_webhook --provisioner my_provisioner --cert-type X509
+'''
+`,
+	}
+}
+
+func updateAction(ctx *cli.Context) (err error) {
+	if err := errs.NumberOfArguments(ctx, 1); err != nil {
+		return err
+	}
+
+	args := ctx.Args()
+
+	provisionerName := ctx.String("provisioner")
+
+	client, err := newCRUDClient(ctx, ctx.String("ca-config"))
+	if err != nil {
+		return err
+	}
+
+	prov, err := client.GetProvisioner(ca.WithProvisionerName(provisionerName))
+	if err != nil {
+		return err
+	}
+	var wh *linkedca.Webhook
+	for _, pwh := range prov.Webhooks {
+		if pwh.Name == args.Get(0) {
+			wh = pwh
+			break
+		}
+	}
+	if wh == nil {
+		return errors.New("provisioner does not have a webhook with that name")
```

Comment:
Maybe state the name of the provisioner and webhook in the error msg?

e.g., `provisioner 'my-provisioner' does not have a webhook with the name 'my-webook'`. This is more a style question, so up to your disgression.

Reply:
97eef2d